Automattic/simplenote-ios
# Fix widget note or note list leakage when pin lock is enabled

### Fix

It's possible to see all your notes from widgets even though pin lock(password) is active.

1. Turn on pin lock.
2. When choose note widget or note list in widgets center, the snapshot of note or note list is shown without protected.
3. When display a note widget or note list widget, the note or note list is shown without protected.
4. When edit a note widget, all note list can be seen, and any note can be selected then displayed in the note widget.

In this PR I have added pinLockIsEnabled state to check all notes or note list in widgets, just like the way when logged out. When pinLockIsEnabled is true, the note or note list is hidden, otherwise it is displayed.

fixes #1445 

### Test

1. Turn on pin lock.
2. When choose note widget or note list in widgets center, the snapshot of note or note list will be shown as "Unlock to see your note/note list".
3. When a note widget or note list widget added, the note or note list will be shown as "Unlock to see your note/note list".
5. When edit a note widget, the note list will not be shown and selected.
6. Turn off pin lock, the note widget or note list widget will be shown normally.

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> `RELEASE-NOTES.txt` was updated in d3adb3ef with:
> 
> > Added markdown support
